components/script/dom/htmlscriptelement.rs

```diff
@@ -159,8 +159,10 @@ impl FetchResponseListener for ScriptContext {
 
     fn process_request_eof(&mut self) {} // TODO(KiChjang): Perhaps add custom steps to perform fetch here?
 
-    fn process_response(&mut self, metadata: Result<Metadata, NetworkError>) {
-        self.metadata = metadata.ok();
+    fn process_response(&mut self,
+                        metadata: Result<Metadata, NetworkError>,
+                        actual_metadata: Option<Result<Metadata, NetworkError>>) {
+        self.metadata = actual_metadata.unwrap_or(metadata).ok();
```

Comment:
With my proposed design, we could use
```rust
self.metadata = metadata.ok().map(|r| match r {
    FetchResponse::Unfiltered(m) => m,
    FetchResponse::Filtered(f) => f.unsafe_response(),
};
```
which matches step 4 of https://html.spec.whatwg.org/multipage/webappapis.html#fetch-a-classic-script .